src/hpke/mod.rs

```diff
@@ -29,44 +29,32 @@ type IpaKem = hpke::kem::X25519HkdfSha256;
 type IpaAead = hpke::aead::AesGcm128;
 type IpaKdf = hpke::kdf::HkdfSha256;
 
-/// Right now we assume the match keys to be 40 bits long. If it is not the case, the decryption
-/// will fail. This assumption allows to keep the bitstrings on the stack, for dynamically sized
-/// match keys we would have to heap allocate.
-type XorReplicated = AdditiveShare<Gf40Bit>;
-
 pub type IpaPublicKey = <IpaKem as hpke::kem::Kem>::PublicKey;
 pub type IpaPrivateKey = <IpaKem as hpke::kem::Kem>::PrivateKey;
 
 pub use hpke::{Deserializable, Serializable};
 
-/// match key size, in bytes
-const MATCHKEY_LEN: usize = <XorReplicated as IpaSerializable>::Size::USIZE;
-
-/// Total len in bytes for an encrypted matchkey including the authentication tag.
-pub const MATCHKEY_CT_LEN: usize =
-    MATCHKEY_LEN + <AeadTag<IpaAead> as Serializable>::OutputSize::USIZE;
-
-pub trait MatchKeyCrypt: GaloisField + IpaSerializable {
+pub trait FieldShareCrypt: GaloisField + IpaSerializable {
```

Comment:
Can we rename it to something better? Just SecretShareCrypt? Not sure if "field" in the name makes it a bit confusing?